packages/test/test-end-to-end-tests/src/test/data-virtualization/snapshotTestCache.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ISnapshot } from "@fluidframework/driver-definitions/internal";
+import {
+	getKeyForCacheEntry,
+	type ICacheEntry,
+	type IFileEntry,
+	type IPersistedCache,
+} from "@fluidframework/odsp-driver-definitions/internal";
+
+export interface ValueWithSnapshot {
+	value: ISnapshot;
+}
+
+export class SnapshotCache implements IPersistedCache {
+	private readonly cache = new Map<string, ValueWithSnapshot>();
+	private readonly versionCache = new Map<string, ValueWithSnapshot>();
+	private readonly versionToCacheKey = new Map<string, string>();
+	public async get(entry: ICacheEntry): Promise<ValueWithSnapshot | undefined> {
+		const key = getKeyForCacheEntry(entry);
+		return this.cache.get(key);
+	}
+	public async put(entry: ICacheEntry, value: ValueWithSnapshot): Promise<void> {
+		const key = getKeyForCacheEntry(entry);
+		this.cache.set(key, value);
+		const versionKey = `${value.value.snapshotTree.id}`;
+		this.versionCache.set(versionKey, value);
+		this.versionToCacheKey.set(versionKey, key);
+	}
+	public async removeEntries(file: IFileEntry): Promise<void> {
+		[...this.cache]
+			.filter(([cacheKey]) => {
+				const docIdFromKey = cacheKey.split("_");
+				return docIdFromKey[0] === file.docId;
+			})
+			.map(([cacheKey]) => {
+				this.cache.delete(cacheKey);
```

Comment:
Don't we need to remove this from other 2 caches that you are maintaining?

Reply:
No, we want to keep it in the other cache so that we can move the summary back in.

The point is I want to be able to load a container with a newer summary do some stuff, and then load a container with a newer summary and do some stuff